nix/process-compose/default.nix

```diff
@@ -53,7 +53,7 @@ in
 
             ${preHook}
 
-            process-compose -p ${toString port} "$@"
+            process-compose ${if builtins.isBool uds then if uds then "-U" else "-p ${toString port}" else "--unix-socket ${uds}"} "$@"
```

Comment:
Why does process-compose allow specifying both `-p` and `-U` and yet here we preclude that possibility?

I'm trying to understand the actual semantics of these options, so as to ascertain whether we are doing the right hing.

@szucsitg

Reply:
i went ahead and implemented the feature, supplying both independently because it could a valid scenario. but if i set the UDS, it disables automatically the TCP port based serving, which i could accept that it would not be the desired behavior for some and might warrant an issue at PC. however for me this was the scenario i was expecting, so i'm fine with it.

i checked PC's code and i think this is happening because of the UDS path provides an early return:
https://github.com/F1bonacc1/process-compose/blob/v1.2.0/src/cmd/root.go#L139-L140

Reply:
Okay. 

Upstream should have gone with an enum CLI option type, but let's work with what they have.

I feel that we should probably do the right thing here; ie., define an enum type ourselves. How about this:

```nix
  let tcpModule = types.submodule { options.port = ...;  };
  let udpModule = types.submodule { options.uds = types.either bool string; };
  ...
  cli.server = types.either tcpModule udpModule;
```

Then, the user can do either `cli.server.port = 8080;` or `cli.server.udp = true;` but not both.

Reply:
i implemented the change, but it doesn't work. i can set the the port and it seems to be working correctly, but uds cannot be set. i'd appreciate some help what am i doing wrong. thanks 😄

Reply:
it's not prettiest but it does work. i tried to do differently by adding check to the submodule, but couldn't figure it out to do it.

Reply:
i simplified the logic, it's easier to follow now

Reply:
@szucsitg Okay, I'll refactor this now before merging.

For posterity, please do not force-push commits.

Reply:
Thanks for updating my solution, your design is definitely better. At least I learned a new pattern 🙂
And sorry for the force push, I'll plan to open one more PR here and some at services flake, I'll keep that in consideration.